packages/flutter/lib/src/material/ink_splash.dart

```diff
@@ -75,6 +75,50 @@ class _InkSplashFactory extends InteractiveInkFeatureFactory {
   }
 }
 
+abstract class InkSplashRippleMixin {
+  // This class is intended to be used as a mixin, and should not be
+  // extended directly.
+  // Reduces duplicate code between InkSplash and InkRipple 
+  factory InkSplashRippleMixin._() => null;
+
+  /// paints a ripple/splash Ink Circle
+  void paintInkCircle(
```

Comment:
Since this mixin provides `paintInkCircle`, I'd recommend just name it as `InkCircleMixin`.

Even simpler, maybe we can just put this `paintInkCircle` as a `@protected` method inside `InteractiveInkFeature` as it's the parent class of both `InkRipple` and `InkSplash`. That will save us the trouble of coming up a name for the new class/mixin.